test/testtriag.jl

```diff
@@ -314,12 +323,10 @@ function test_triangular(elty1_types)
             @test ((A1 \ A1)::t1) ≈ M1 \ M1
 
             # Begin loop for second Triangular matrix
-            @testset for elty2 in (Float32, Float64, BigFloat, ComplexF32, ComplexF64, Complex{BigFloat}, Int)
-                # Only test methods for the same element type and a single combination of mixed element types
-                # to avoid too much compilation
-                if !(elty1 == elty2 || elty1 ∈ (ComplexF32, Int) || elty1 ∈ (ComplexF32, Int))
```

Comment:
How much of a difference does it make to just fix the bug here, i.e. to change the last `elty1` to `elty2`?

I'm not sure it is a good idea to test against a random eltype. It can be quite confusing with randomness in tests.

Reply:
The seed is fixed, so the tests are fully reproducible. The randomness only ensures that different types are chosen in the tests.

Changing the `elty1` to `elty2` here still leads to `Int` being tested against every type, and also tests the floating-point types against `Int` and `ComplexF32`. This increases the test runtime by quite a bit. This PR changes this to test each type against only two types: itself, and one other type.

Reply:
Please see https://github.com/JuliaLang/LinearAlgebra.jl/pull/1294#issuecomment-2818682552 for details

Reply:
> Changing the elty1 to elty2 here still leads to Int being tested against every type

That is because I wrote the condition incorrectly. It doesn't match the comment above it. The condition should be
```julia
if !(elty1 == elty2 || (elty1 ∈ (ComplexF32, Int) && elty2 ∈ (ComplexF32, Int)))
```

> The seed is fixed, so the tests are fully reproducible. The randomness only ensures that different types are chosen in the tests.

I see. I think fixing the conditions as above would be better. The idea is that the combination of `Int` and `ComplexF32` reaches most corner cases.

Reply:
Oh wait, doesn't this mean that we test against every `elty2` for `elty1 == Int`?